src/imgui/KeyMappings.hh

```diff
@@ -0,0 +1,171 @@
+	{ImGuiKey_C,              SDLK_c},
+	{ImGuiKey_D,              SDLK_d},
+	{ImGuiKey_E,              SDLK_e},
+	{ImGuiKey_F,              SDLK_f},
+	{ImGuiKey_G,              SDLK_g},
+	{ImGuiKey_H,              SDLK_h},
+	{ImGuiKey_I,              SDLK_i},
+	{ImGuiKey_J,              SDLK_j},
+	{ImGuiKey_K,              SDLK_k},
+	{ImGuiKey_L,              SDLK_l},
+	{ImGuiKey_M,              SDLK_m},
+	{ImGuiKey_N,              SDLK_n},
+	{ImGuiKey_O,              SDLK_o},
+	{ImGuiKey_P,              SDLK_p},
+	{ImGuiKey_Q,              SDLK_q},
+	{ImGuiKey_R,              SDLK_r},
+	{ImGuiKey_S,              SDLK_s},
+	{ImGuiKey_T,              SDLK_t},
+	{ImGuiKey_U,              SDLK_u},
+	{ImGuiKey_V,              SDLK_v},
+	{ImGuiKey_W,              SDLK_w},
+	{ImGuiKey_X,              SDLK_x},
+	{ImGuiKey_Y,              SDLK_y},
+	{ImGuiKey_Z,              SDLK_z},
+	{ImGuiKey_F1,             SDLK_F1},
+	{ImGuiKey_F2,             SDLK_F2},
+	{ImGuiKey_F3,             SDLK_F3},
+	{ImGuiKey_F4,             SDLK_F4},
+	{ImGuiKey_F5,             SDLK_F5},
+	{ImGuiKey_F6,             SDLK_F6},
+	{ImGuiKey_F7,             SDLK_F7},
+	{ImGuiKey_F8,             SDLK_F8},
+	{ImGuiKey_F9,             SDLK_F9},
+	{ImGuiKey_F10,            SDLK_F10},
+	{ImGuiKey_F11,            SDLK_F11},
+	{ImGuiKey_F12,            SDLK_F12},
+	{ImGuiKey_F13,            SDLK_F13},
+	{ImGuiKey_F14,            SDLK_F14},
+	{ImGuiKey_F15,            SDLK_F15},
+	{ImGuiKey_F16,            SDLK_F16},
+	{ImGuiKey_F17,            SDLK_F17},
+	{ImGuiKey_F18,            SDLK_F18},
+	{ImGuiKey_F19,            SDLK_F19},
+	{ImGuiKey_F20,            SDLK_F20},
+	{ImGuiKey_F21,            SDLK_F21},
+	{ImGuiKey_F22,            SDLK_F22},
+	{ImGuiKey_F23,            SDLK_F23},
+	{ImGuiKey_F24,            SDLK_F24},
+	{ImGuiKey_AppBack,        SDLK_AC_BACK},
+	{ImGuiKey_AppForward,     SDLK_AC_FORWARD},
+	// ... order doesn't matter  (this table does not end up in the final executable)
+});
+
+[[nodiscard]] SDL_Keycode ImGuiKey2SDL(ImGuiKey imgui)
+{
+	static constexpr auto lookup = []{
+	    auto result = unsortedKeys;
+	    std::ranges::sort(result, {}, &ImGui_SDL_Key::imgui);
+	    return result;
+	}();
+	auto it = std::ranges::lower_bound(lookup, imgui, {}, &ImGui_SDL_Key::imgui);
+	return ((it != lookup.end()) && (it->imgui == imgui)) ? it->sdl : SDLK_UNKNOWN;
+}
+
+[[nodiscard]] ImGuiKey SDLKey2ImGui(SDL_Keycode sdl)
+{
+	static constexpr auto lookup = []{
+	    auto result = unsortedKeys;
+	    std::ranges::sort(result, {}, &ImGui_SDL_Key::sdl);
+	    return result;
+	}();
+	auto it = std::ranges::lower_bound(lookup, sdl, {}, &ImGui_SDL_Key::sdl);
+	return ((it != lookup.end()) && (it->sdl == sdl)) ? it->imgui : ImGuiKey_None;
+}
+
+// Note: for openMSX replace  (until we can use clang-16 everywhere):
+//    std::ranges::sort         ->   ranges::sort
+//    std::ranges::lower_bound  ->   ranges::lower_bound
+//    #include <algorithm>      ->   #include "ranges.hh"
```

Comment:
These comment lines (and some above) were meant as an explanation on the compiler-explorer prototype.
They were meant as instructions for you, for when you were going to use this in the openMSX code ;-)

(This version will break with clang-15 because it doesn't have `std::ranges::sort()` yet. The solution is to instead use our `ranges::sort()` function.)